filters/HAGFilter.cpp

```diff
@@ -99,13 +139,94 @@ void HAGFilter::filter(PointView& view)
+                auto y = gView->getFieldAs<double>(Dimension::Id::Y, ids[j]);
+                auto z = gView->getFieldAs<double>(Dimension::Id::Z, ids[j]);
+                auto distance = std::sqrt(std::pow(x - x0, 2) + std::pow(y - y0, 2));
+                if (distance == 0) {
+                    exact_match = true;
+                    z1 = z;
+                    break;
+                }
+                if (m_max_distance > 0 && distance > m_max_distance) {
+                    break;
+                } else {
+                    auto weight = 1 / std::pow(distance, 2);
+                    weights += weight;
+                    z_accumulator += weight * z;
+                }
+                if (x > max_x) {
+                    max_x = x;
+                }
+                if (x < min_x) {
+                    min_x = x;
+                }
+                if (y > max_y) {
+                    max_y = y;
+                }
+                if (y < min_y) {
+                    min_y = y;
+                }
+            }
+            if (exact_match || m_allow_extrapolation || (x0 > min_x && x0 < max_x && y0 > min_y && y0 < max_y)) {
+                z1 = z_accumulator / weights;
+            }
+        } else if (m_delaunay == true) { // Delaunay-based interpolation
+            auto neighbors = std::vector<double>();
+
+            for(size_t j = 0; j < ids.size(); ++j)
+            {
+                neighbors.push_back(gView->getFieldAs<double>(Dimension::Id::X, ids[j]));
+                neighbors.push_back(gView->getFieldAs<double>(Dimension::Id::Y, ids[j]));
+            }
+            auto triangulation = delaunator::Delaunator(neighbors);
+            auto triangles = triangulation.triangles;
+
+            z1 = std::numeric_limits<double>::infinity();
+            for (size_t j = 0; j < triangles.size(); j += 3)
+            {
+                auto ai = triangles[j+0];
+                auto bi = triangles[j+1];
+                auto ci = triangles[j+2];
+                double ax = gView->getFieldAs<double>(Dimension::Id::X, ids[ai]);
+                double ay = gView->getFieldAs<double>(Dimension::Id::Y, ids[ai]);
+                double az = gView->getFieldAs<double>(Dimension::Id::Z, ids[ai]);
+                double bx = gView->getFieldAs<double>(Dimension::Id::X, ids[bi]);
+                double by = gView->getFieldAs<double>(Dimension::Id::Y, ids[bi]);
+                double bz = gView->getFieldAs<double>(Dimension::Id::Z, ids[bi]);
+                double cx = gView->getFieldAs<double>(Dimension::Id::X, ids[ci]);
+                double cy = gView->getFieldAs<double>(Dimension::Id::Y, ids[ci]);
+                double cz = gView->getFieldAs<double>(Dimension::Id::Z, ids[ci]);
+
+                z1 = std::min(z1, distance_along_z(ax, ay, az, bx, by, bz, cx, cy, cz, x0, y0));
+            }
+            if (z1 == std::numeric_limits<double>::infinity())
```

Comment:
I don't understand this.  It seems we're initializing z1 with infinity.  Then I would think the minimum of infinity and anything else would be anything else.  If I'm not misunderstanding, simpler logic would just be to assign to z1 as:

```
std::min(
    gView->getFieldAs<double>(Dimension::Id::Z, ids[0]),
    distance_along_z(ax, ay, az, bx, by, bz, cx, cy, cz, x0, y0)
);